### DISCUSSION TITLE
/auth/sign_in results in 422 error on new install

### DISCUSSION THREAD

**Discussion by BenLand100** [mastodon/mastodon#22699] [Q&A] [answered]
I've followed the "install from source" instructions to setup a mastodon instance on gentoo linux using apache as a reverse proxy. Nominally this works, except the instance is fully unusable! :) If I try to login, I get an 422 error from puma on `/auth/sign_in`.

```
[23e05cb8-75cf-4267-ae7e-20f2371076bc] method=POST path=/auth/sign_in format=html controller=Auth::SessionsController action=create status=422 duration=20.04 view=7.56 db=3.52
```

I've fiddled with everything I can think of to try to understand what's going wrong here, but the error is completely and totally opaque, even with `--debug -v` enabled with puma. Can anyone tell me what might be going wrong?

**Comment by BenLand100** [comment, marked as answer]
The culprit appears to be:
```
Header always set Referrer-Policy "no-referrer"
```
Is it understood why mastodon breaks (with no reasonable error message) when this pretty standard header is set?

**Reply by verymilan** [reply]
Thanks a ton, this really was a well hidden trap and i found this here discussion way too late...
